src/lib/support/Pool.h

```diff
@@ -331,6 +394,86 @@ class HeapObjectPool : public internal::Statistics, public HeapObjectPoolExitHan
 #endif // __SANITIZE_ADDRESS__
     }
 
+    /// Provides iteration over active objects in the pool.
+    ///
+    /// NOTE: There is extra logic to allow objects release WHILE the iterator is
+    ///       active while still allowing to advance the iterator.
+    ///       This is done by flagging an iteration depth whenever an active
+    ///       iterator exists. This also means that while a pool iterator exists, releasing
+    ///       of tracking memory objects may be deferred until the last active iterator is
+    ///       released.
+    class ActiveObjectIterator
+    {
+    public:
+        using value_type = T;
+        using pointer    = T *;
+        using reference  = T &;
+
+        ActiveObjectIterator() {}
+        ActiveObjectIterator(const ActiveObjectIterator & other) : mCurrent(other.mCurrent), mEnd(other.mEnd)
+        {
+            if (mEnd != nullptr)
+            {
+                mEnd->mIterationDepth++;
+            }
+        }
+
+        ActiveObjectIterator & operator=(const ActiveObjectIterator & other)
+        {
+            if (mEnd != nullptr)
+            {
+                mEnd->mIterationDepth--;
```

Comment:
I will add some comments. Apparently `ForEachActiveObject` guaranteed that Release() is possible while iterating (for example for expiring things).

This would not have been needed if you use iterators as most iterators say "invalidated once you remove" however I apparently had to support this and used the same method that ForEachActiveObject used (this iteration depth).